ports/stm32/mpbthciport.c

```diff
@@ -35,23 +36,61 @@
 
 #if MICROPY_PY_BLUETOOTH
 
-#define DEBUG_printf(...) // printf(__VA_ARGS__)
+#define DEBUG_printf(...) // printf("mpbthciport.c: " __VA_ARGS__)
 
 uint8_t mp_bluetooth_hci_cmd_buf[4 + 256];
 
 // Must be provided by the stack bindings (e.g. mpnimbleport.c or mpbtstackport.c).
+// Request new data from the uart and pass to the stack, and run pending events/callouts.
 extern void mp_bluetooth_hci_poll(void);
 
 // Hook for pendsv poller to run this periodically every 128ms
-#define BLUETOOTH_HCI_TICK(tick) (((tick) & ~(SYSTICK_DISPATCH_NUM_SLOTS - 1) & 0x7f) == 0)
+#define BLUETOOTH_HCI_TICK(tick) (((tick) & ~(SYSTICK_DISPATCH_NUM_SLOTS - 1) & 0xf7) == 0)
+
+#if MICROPY_PY_BLUETOOTH_USE_SYNC_EVENTS
+
+// For synchronous mode, we run all BLE stack code inside a scheduled task.
+// This task is scheduled periodically (every 128ms) via SysTick, or
+// immediately on HCI UART RXIDLE.
+
+STATIC mp_obj_t run_events_scheduled_task(mp_obj_t none_in);
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(run_events_scheduled_task_obj, run_events_scheduled_task);
```

Comment:
the decl of run_events_scheduled_task is not needed, and run_events_scheduled_task_obj can move to its standard place below run_events_scheduled_task defn